scripts/dts/edtlib.py

```diff
@@ -1255,6 +1262,10 @@ def _check_binding(binding, binding_path):
             _err("enum in {} for property '{}' is not a list"
                  .format(binding_path, prop_name))
 
+        if "const" in options and isinstance(options["const"], (int, str)):
+           _err("const in {} for property ({}) is not a scalar"
```

Comment:
`"'const' value for property '{}' in {} is not a scalar"` might be better, since the name is essential info.

('' around const to make it clearer that it's a keyword.)